### PR TITLE
E2e tests

### PR DESCRIPTION
**Pull request by gretelliz**
This PR is to test the return codes.

### CODE REVIEW

**Review by italovalcy — COMMENTED** [review]
Hi @gretelliz as we discussed on today's meeting, can you please refactor this test and split the return code validation function into a new file, let's say `test_06_l2vpn_return_codes.py` ?

**Comment by gretelliz** [comment]
Sure @italovalcy, please note that I did what you suggested in the last commit.

**Comment by italovalcy** [comment]
Hi Gretel I see that some of the changes described on this PR are also listed on PR #8. I believe that is because the target branch here is also main, while the branch you are using here was based on the previous one from PR #8. I will change the target branch here to facilitate the review, okay?

File: tests/test_06_l2vpn_return_codes.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        201: L2VPN Service Created
+        P2P with VLAN translation
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        payload = {
+            "name": "VLAN between AMPATH/100 and TENET/150",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "100",},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "150",},
+            ],}
```
Comment (italovalcy) [inline tests/test_06_l2vpn_return_codes.py:40]:
```suggestion
            ]
        }
```

File: tests/test_06_l2vpn_return_codes.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    def setup_class(cls):
+        cls.net = NetworkTest(["ampath", "sax", "tenet"])
+        cls.net.wait_switches_connect()
+        cls.net.run_setup_topo()
+
+    @classmethod
+    def teardown_class(cls):
+        cls.net.stop()
+    
+    def test_010_code201(self):
```
Comment (italovalcy) [inline tests/test_06_l2vpn_return_codes.py:28]:
```suggestion
    def test_010_create_l2vn_with_vlan_id_code201(self):
```

File: tests/test_06_l2vpn_return_codes.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+        payload = {
+            "name": "VLAN between AMPATH/100 and TENET/150",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "100",},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "150",},
+            ],}
+        response = requests.post(api_url, json=payload)
+        assert response.status_code == 201, response.text
+    
+    def test_011_code201(self):
```
Comment (italovalcy) [inline tests/test_06_l2vpn_return_codes.py:44]:
```suggestion
    def test_011_create_l2vn_with_vlan_any_code201(self):
```

File: tests/test_06_l2vpn_return_codes.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+        201: L2VPN Service Created
+        P2P with option "any"
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        payload = {
+            "name": "VLAN between AMPATH/300 and TENET/any",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "300",},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "any",},
+            ],}
```
Comment (italovalcy) [inline tests/test_06_l2vpn_return_codes.py:56]:
```suggestion
            ]
        }
```

**Review by italovalcy — CHANGES_REQUESTED** [review]
Hi Gretel, amazing to see how this is progressing well! Congratulations! I have two comments for you please to consider:
1. Can you please make a slight change on the name of the tests to add a very short description about what the test is doing? I added suggestions for the two first ones, but please also apply this change on the others!
2. Can you please run a pylint or similar on this code? I see a few commas on the wrong place and pylint will help you identify some improvements on the formatting style.

File: tests/test_06_l2vpn_return_codes.py [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+        Test the return code for creating a SDX L2VPN
+        201: L2VPN Service Created
+        P2P with VLAN translation
+        """
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        payload = {
+            "name": "VLAN between AMPATH/100 and TENET/150",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "100",},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "150",},
```
Comment (italovalcy) [inline tests/test_06_l2vpn_return_codes.py:39]:
```suggestion
                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "100"},
                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "150"},
```

**Comment by gretelliz** [comment]
Thank you for your review @italovalcy.

**Review by italovalcy — APPROVED** [review]
Very nicely done, Gretel! LGTM!

**Pull request merged**